BL-4633 Fix crash in GetPageBody()

gmartin7 opened the pull request:
Agatha got a "Window does not have a global JSObject" exception from the Geckofx-Core. This prevents that exception crashing Bloom. Comments in the code suggest that a failure here only indicates a timing problem that keeps Bloom from saving the zoom state.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomdesktop/1669)
<!-- Reviewable:end -->

JohnThomson commented:
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/Edit/EditingView.cs, line 982 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1669#-KkBMcBOm0qA4H6lbquW:-KkBMcBOm0qA4H6lbquX:b-6rk14m) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/e51ffdf59cb8f5947c7ed318a282ff91adf97fde/src/BloomExe/Edit/EditingView.cs#L982)):*
> ```Smalltalk
> 		}
> 
> 		public GeckoElement GetPageBody()
> ```

I suppose you've checked callers to be sure that no more serious problem will be caused when this fails and that all callers can handle a null result?
I wonder whether this is worth logging, not just Debug.Writeline?
A comment describing circumstances in which it might fail seems appropriate.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1669)*
<!-- Sent from Reviewable.io -->

gmartin7 commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/Edit/EditingView.cs, line 982 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1669#-KkBMcBOm0qA4H6lbquW:-KkCbk5She8t6odcRki0:b-78shft) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/e51ffdf59cb8f5947c7ed318a282ff91adf97fde/src/BloomExe/Edit/EditingView.cs#L982)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

I suppose you've checked callers to be sure that no more serious problem will be caused when this fails and that all callers can handle a null result?
I wonder whether this is worth logging, not just Debug.Writeline?
A comment describing circumstances in which it might fail seems appropriate.
</blockquote></details>

Yes, the only caller says:
			var body = _view.GetPageBody();
			Debug.Assert(body!=null, "(Debug Only) no body when doing SavePageFrameState()" );

			// not worth crashing over a timing problem that means we don't save zoom state
			if (body == null)
				return; // BL-3075, not sure how this can happen but it has. Possibly the view is in some state like about:null which has no body.

I've changed the Debug.Writeline to a Logger.WriteEvent and added a method comment that mentions the only caller.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1669)*
<!-- Sent from Reviewable.io -->

JohnThomson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1669#-:-KkCu4waYRX7CnNOaqX9:bnfp4nl)*
<!-- Sent from Reviewable.io -->
